src/utils/pathEncoding.tsx

```diff
@@ -0,0 +1,48 @@
+import { generatePath } from 'react-router-dom';
+
+import { NotFoundError } from '@/modules/errors/util';
+import IdEncoder from '@/modules/hmis/IdEncoder';
+import { HmisEnums } from '@/types/gqlEnums';
+
+export const isIdParam = (key: string) => key.match(/Id$/);
+
+const enumParams: Record<string, string[]> = {
+  formRole: Object.keys(HmisEnums.FormRole),
+};
+
```

Comment:
maybe some docs here would be helpful
```suggestion
/**
 * Decodes encrypted ID parameters
 * Converts enum parameter values to uppercase and validates them
 * Throws a NotFoundError for unrecognized enum parameter values
 */
```